src/node.rs

```diff
@@ -69,6 +69,21 @@ pub struct Node {
     pub material: Option<Material>,
 }
 
+impl<'a> From<&'a NodeInternal> for Node {
```

Comment:
any reason this is not just a `to_node(&self)` method in `NodeInternal`?

Reply:
No particular reason except for the common way of expressing translation from one type to another using `From` trait. Any reason not to do that?